### PR TITLE
Optional Replication Url

### PR DESCRIPTION
**Pull request by TurtIeSocks**
- Rewrite of #258
- Related to #257
- Somewhat related to #269

-------------------------
- Throws error if a REPLICATION_UPDATE_INTERVAL or REPLICATION_RECHECK_INTERVAL is set and the REPLICATION_URL is not set
- Skips replication init
- Readme addition noting when REPLICATION URL is required
- Readme addition providing an example docker command without it
- GitHub CI test

### CODE REVIEW

**Review by leonardehrenfried — APPROVED** [review]

**Comment by leonardehrenfried** [comment]
This looks really nice, well done. Could you please exclude the failing test for 3.6?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -148,3 +148,13 @@ jobs:
              -t mediagis/nominatim:${{ matrix.nominatim.version }} \
              -t mediagis/nominatim:${{ matrix.nominatim.version }}-${{ github.sha }} .
         working-directory: ${{ matrix.nominatim.version }}
+
+      - name: Check when REPLICATION_URL is blank
```
Comment (leonardehrenfried) [inline .github/workflows/ci.yml:152]: Can you move this up please, so it's run before the docker build and push?

Comment (TurtIeSocks) [reply .github/workflows/ci.yml]: Np, all set in latest commit.

**Review by leonardehrenfried — APPROVED** [review]
This looks ready to be merged but maybe @philipkozeny wants to have a final look.

Thanks for the contribution!

**Comment by TurtIeSocks** [comment]
> This looks ready to be merged but maybe @philipkozeny wants to have a final look.
> 
> Thanks for the contribution!

Of course, thank you for the work you and the other maintainers have put into this repo! The community of devs I work with have been using these docker containers for a number of years 😃

**Comment by philipkozeny** [comment]
Thakns for the great PR @TurtIeSocks !

**Pull request merged**